lambdas/process-cri-callback/src/test/java/uk/gov/di/ipv/core/processcricallback/pact/bavCri/ContractTest.java

```diff
@@ -0,0 +1,655 @@
+    // sent by the CRI team
+    private static final String VALID_VC_HEADER =
+            """
+            {
+              "typ": "JWT",
+              "alg": "ES256"
+            }
+            """;
+    // 2099-01-01 00:00:00 is 4070908800 in epoch seconds
+    private static final String VALID_BAV_VC_BODY =
+            """
+            {
+              "sub": "test-subject",
+              "iss": "dummyBavComponentId",
+              "nbf": 4070908800,
+              "exp": 4070909400,
+              "vc": {
+                "evidence": [
+                  {
+                    "validityScore": 2,
+                    "strengthScore": 3,
+                    "txn": "dummyTxn",
+                    "type": "IdentityCheck"
+                  }
+                ],
+                "credentialSubject": {
+                  "bankAccount": [
+                    {
+                      "accountNumber": "12345678",
+                      "sortCode": "103233"
+                    }
+                  ],
+                  "name": [
+                    {
+                      "nameParts": [
+                        {
+                          "type": "GivenName",
+                          "value": "Kenneth"
+                        },
+                        {
+                          "type": "FamilyName",
+                          "value": "Decerqueira"
+                        }
+                      ]
+                    }
+                  ],
+                  "birthDate": [
+                    {
+                      "value": "1962-10-11"
+                    }
+                  ]
+                },
+                "jti": "dummyJti"
+                }
+            }
+            """;
+    // If we generate the signature in code it will be different each time, so we need to generate a
+    // valid signature (using https://jwt.io works well) and record it here so the PACT file doesn't
+    // change each time we run the tests.
+    private static final String VALID_BAV_VC_SIGNATURE =
+            "ARHudEXt_OpGqJPP603JvjjmykiibzIUiKD9YvaqLlFAokimzRVDxS3lmu19UMpumLxdXxyz_MMb0ZyB3xgo-A";
+
+    private static final String FAILED_BAV_VC_BODY =
+            """
+                {
+                   "sub": "test-subject",
+                   "iss": "dummyBavComponentId",
+                   "nbf": 4070908800,
+                   "exp": 4070909400,
+                   "vc": {
+                     "evidence": [
+                       {
+                         "failedCheckDetails": [
+                           {
+                             "identityCheckPolicy": "none",
+                             "checkMethod": "data"
+                           }
+                            ],
+                         "validityScore": 0,
+                         "strengthScore": 3,
+                         "ci": [
+                           "D02"
```

Comment:
Is `D02` a real CI BAV can return? The stub data has `D15` for BAV

Reply:
`D15` is the only CI BAV can return to us, however IPV Core cannot handle `D15` as a CI, it believes that it is invalid.

Currently in the journey map, there is no ability to handle CIs with the BAV_CRI, however the validate() method that we are running in our code believes that `D02` is a valid CI.

Not sure really what the best plan of action for this is, but feels like changing the CI handling is perhaps outside of the scope of this ticket.

@blakeyp any thoughts?

Reply:
Not sure that's correct, have you tried changing line 626?

Reply:
Oh damn, I forgot we had to mock that out 🤦